api/server/volume.go

```diff
@@ -2063,3 +2076,19 @@ func GetVolumeAPIRoutesWithAuth(
 	return router, nil
 
 }
+
+func isRequestFromScheduler(r *http.Request) bool {
+	userAgent := r.Header.Get("User-Agent")
```

Comment:
Did we check if any pxctl volume update requests uses this?

Let's do a broad search for all "User-Agent" usages for volume driver calls.

Reply:
Sched driver only used in the following spots in pxctl:
```
grant@ip-70-0-0-129:~/workspace/go/src/github.com/portworx/porx$ grep -nr 'schedVolDriver' px/cli
px/cli/volumes.go:91:   schedVolDriver    volume.VolumeDriver
px/cli/volumes.go:531:                  volumes, err = schedVolDriver.Enumerate(locator, nil)
px/cli/volumes.go:618:          volumes, err = schedVolDriver.Enumerate(locator, nil)
px/cli/volumes.go:1262: allVolumes, err := schedVolDriver.Inspect(d)
px/cli/volumes.go:1752: err = schedVolDriver.Delete(vol.Id)
px/cli/volumes.go:3814: schedVolDriver = volumeclient.VolumeDriver(schedClient)
```

Reply:
and to be safe, we don't seem to be used a sched driver anywhere else in px:
```
px/cli/volumes.go:90:   volDriver         volume.VolumeDriver
px/cli/volumes.go:91:   schedVolDriver    volume.VolumeDriver
Binary file px/pxctl/pxctl/test/alerts/server/server matches
Binary file px/pxctl/pxctl/test/pxctl matches
px/sched/filter/k8s/k8s.go:145: driver volume.VolumeDriver,
px/sched/filter/k8s/k8s.go:531: driver volume.VolumeDriver,
px/sched/filter/k8s/k8s.go:753:func (f *k8sFilter) PreDelete(volumeID string, driver volume.VolumeDriver) error {
px/sched/filter/k8s/k8s.go:758:func (f *k8sFilter) isVolumeBound(volumeID string, driver volume.VolumeDriver) error {
px/sched/filter/k8s/k8s.go:880:func getRemoteVolDriver(host string) (volume.VolumeDriver, error) {
px/sched/filter/k8s/k8s.go:891:func getSourceVolByName(srcName string, driver volume.VolumeDriver) (string, error) {
px/sched/filter/filter.go:17:           driver volume.VolumeDriver,
px/sched/filter/filter.go:23:   PreDelete(volumeID string, driver volume.VolumeDriver) error
px/sched/filter/filter.go:56:           driver volume.VolumeDriver,
px/sched/filter/filter.go:76:   _ *api.VolumeSpec, _ volume.VolumeDriver) (stor_api.VolumeAction, error) {
px/sched/filter/filter.go:86:func (f *BaseSchedFilter) PreDelete(_ string, driver volume.VolumeDriver) error {
px/sched/filter/filter.go:155:  _ volume.VolumeDriver,
px/test/helper/volume/managers/osd/osd.go:21:   volumeDriver     []osdvolume.VolumeDriver
px/test/helper/volume/managers/osd/osd.go:54:   volumeDriver := make([]osdvolume.VolumeDriver, nodes)
```

Reply:
places where we reference the sched driver name constant 
```
grant@ip-70-0-0-129:~/workspace/go/src/github.com/portworx/porx$ grep -nr 'PxSchedDriverName' .
./px/cli/volumes.go:3808:	schedClient, err := apiclient.GetNewVolumeclient(host, port, stor_api.PxSchedDriverName)
./px/sched/filter/k8s/k8s.go:881:	clnt, e := apiclient.GetNewSystemAuthVolumeclient(host, config.GetStringPort(config.PxMgmtPort), stor_api.PxSchedDriverName)
./px/storage/stor.go:670:		if err = ostvoldriver.Register(stor_api.PxSchedDriverName, nil); err != nil {
./px/storage/stor.go:672:				stor_api.PxSchedDriverName, err)
./px/storage/stor.go:1654:	driverMap[stor_api.PxSchedDriverName] = d
./px/storage/stor.go:1658:		schedDriver, err := ostvoldriver.Get(stor_api.PxSchedDriverName)
./px/storage/stor.go:1660:			logrus.Errorf("Unable to get volume driver for %v: %v", stor_api.PxSchedDriverName, err)
./px/storage/stor.go:1663:		driverMap[stor_api.PxSchedDriverName] = schedDriver
./px/storage/stor.go:1712:	csiStorageDriverName := stor_api.PxSchedDriverName
./px/storage/stor.go:2238:	ostvoldriver.Add(stor_api.PxSchedDriverName,
./storage/api/api.go:621:	// PxSchedDriverName name of the scheduler aware pxd driver
./storage/api/api.go:622:	PxSchedDriverName = "pxd-sched"
./storage/driver/test/volume.go:272:	driverMap[stor_api.PxSchedDriverName] = od
./storage/driver/volume/nfs/ref_v2.go:592:		localDriver, err := ostvoldriver.Get(api.PxSchedDriverName)
./storage/driver/volume/nfs/ref_v2.go:620:		localDriver, err := ostvoldriver.Get(api.PxSchedDriverName)
./storage/driver/volume/node_map.go:973:	return m.proxyVolumeDriver(driver, volumeID, true, false, preferSameClusterDomain, stor_api.PxSchedDriverName)
./storage/tags:2611:PxSchedDriverName	api/api.go	/^	PxSchedDriverName = "pxd-sched"$/;"	c
```

Reply:
these are mostly used for:
1. registering/initializing px
2. proxy attach/detach/mount/umount